ttnn/cpp/ttnn/operations/pool/avgpool/avg_pool.hpp

```diff
@@ -4,7 +4,25 @@
 
 #pragma once
 
-#include "tt_eager/tt_dnn/op_library/pool/average_pool.hpp"
+#include "tt_metal/host_api.hpp"
+#include "tensor/tensor.hpp"
+
+#include "tt_dnn/op_library/operation.hpp"
+
+namespace tt {
+namespace tt_metal {
+
+enum class PoolType {
+    AVG
+};
+
+Tensor average_pool_2d(const Tensor& input, const MemoryConfig& memory_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG, const std::optional<DataType>& output_dtype = std::nullopt);
```

Comment:
We prefer to not use ` memory_config = operation::DEFAULT_OUTPUT_MEMORY_CONFIG` and instead keep it and `optional<MemoryConfig> memory_config = std::nullopt`

Then on the op side, if it is nullopt, we take the memory config from the input. 

Is it possible here?

Reply:
Yes. I will make this change into a separate issue.